services/context-box/server/server.go

```diff
@@ -55,13 +55,25 @@ func (*server) SaveConfigScheduler(ctx context.Context, req *pb.SaveConfigReques
 	// Save new config to the DB
 	config.DsChecksum = config.MsChecksum
 	config.SchedulerTTL = 0
-	err := database.UpdateDs(config)
-	if err != nil {
+	if err := database.UpdateDs(config); err != nil {
 		return nil, fmt.Errorf("error while updating dsChecksum for %s : %w", config.Name, err)
 	}
 
-	err = database.UpdateSchedulerTTL(config.Name, config.SchedulerTTL)
-	if err != nil {
+	if config.DesiredState != nil {
+		// Update workflow state for k8s clusters.
+		for _, cluster := range config.DesiredState.Clusters {
+			if err := database.UpdateWorkflowState(config.Name, cluster.ClusterInfo.Name, config.State[cluster.ClusterInfo.Name]); err != nil {
+				return nil, fmt.Errorf("error while updating workflow state for k8s cluster %s in config %s : %w", cluster.ClusterInfo.Name, config.Name, err)
+			}
+		}
+		// Update workflow state for LB clusters.
+		for _, cluster := range config.DesiredState.LoadBalancerClusters {
+			if err := database.UpdateWorkflowState(config.Name, cluster.ClusterInfo.Name, config.State[cluster.ClusterInfo.Name]); err != nil {
+				return nil, fmt.Errorf("error while updating workflow state for LB cluster %s in config %s : %w", cluster.ClusterInfo.Name, config.Name, err)
+			}
+		}
```

Comment:
LB state isn't tracked, as of now, in builder when the config enters the workflow. The state of the cluster represents both the LB's attached to it and the cluster itself, thus this section is redundant.

Reply:
My bad did not notice that, thanks for pointing this out!